### PR TITLE
Added API functionality

### PR DESCRIPTION
**Pull request by singhcoder694**
Added the functionality of API to render the data from cloudinary. Now as anyone posts data to cloudinary it will automatically will be visible to Resouces pages. Just some of the static things would be missing like image, title etc other things would be availabel there. #1 

## Summary by CodeRabbit

- **New Features**
  - Implemented dynamic resource fetching across various components to provide up-to-date content.

- **Enhancements**
  - Enhanced resource components to include a 'Load More' functionality, displaying additional resources on demand.

- **Refactor**
  - Refactored resource components to use React's `useState` and `useEffect` hooks for improved state management.

- **Bug Fixes**
  - Addressed issues with static resource data by integrating server-side fetching for real-time updates.

### CODE REVIEW

File: src/components/resources/language/index.js [hunk 1]
```diff
@@ -48,96 +48,109 @@ const Description = tw.div``;
...
+    imageSrc:
+      "https://images.unsplash.com/photo-1542831371-d531d36971e6?ixlib=rb-1.2.1&ixid=MXwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHw%3D&auto=format&fit=crop&w=1024&q=80",
+    category: "Book",
+    date: "Feb 22, 2021",
+    title: "OOPs Question",
+    description: "",
+    url: "https://res.cloudinary.com/cuchapter/image/upload/v1656481991/addResources/languages/oops_questions_for_cse_3rd_sem_hj2f4p.pdf",
+    featured: false,
+  },
+]
```
[the remark this answers is not in the document]
Comment (singhcoder694) [reply src/components/resources/language/index.js:132]: It is static Data which will be used to get the static datas like image URL, title etc. which is not present in Cloudinary API call.

File: src/components/resources/database/index.js [hunk 2]
```diff
@@ -48,61 +48,73 @@ const Description = tw.div``;
...
+    fetch("http://localhost:3001/resources/getresources",)
+      .then((res) => {
+        return (res.json());
+      })
+      .then((result)=> {
+        setPost(getArray(result,5,staticData));
+      })
+      .catch((err) => {
+        console.log(err);
+      });
```
[the remark this answers is not in the document]
Comment (singhcoder694) [reply src/components/resources/database/index.js:116]: Will be done in future.

File: src/components/resources/cloud-devops/index.js [hunk 3]
```diff
@@ -48,26 +48,38 @@ const Description = tw.div``;
...
+      .then((res) => {
+        return (res.json());
+      })
+      .then((result)=> {
+        setPost(getArray(result,9,staticData));
+      })
+      .catch((err) => {
+        console.log(err);
+      });
+  },[])
```
[the remark this answers is not in the document]
Comment (singhcoder694) [reply src/components/resources/cloud-devops/index.js:82]: Yes that is done properly in backend.

File: src/components/resources/getResources.js [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+            post.push({
+            imageSrc:posts[drive_arr[i]].imageSrc,
+            title:posts[drive_arr[i]].title,
+            description:posts[drive_arr[i]].description,
+            url: posts[drive_arr[i]].url,
+            category: posts[drive_arr[i]].category,
+            date: posts[drive_arr[i]].date,
+        });
+    }
+    return post;
```
[the remark this answers is not in the document]
Comment (singhcoder694) [reply src/components/resources/getResources.js:56]: 9 is there to provide reference from backend array. post[0] is there is there is no image URL present for now that can be placed just for better look.

File: src/components/resources/ai/index.js [hunk 5]
```diff
@@ -8,7 +8,7 @@ import Header from "components/headers/light.js";
 import Footer from "components/footers/footers.js";
 import { SectionHeading } from "components/misc/Headings";
 import { PrimaryButton } from "components/misc/Buttons";
-
+import { getArray } from "../getResources";
```
Comment (ank1traj) [inline src/components/resources/ai/index.js:11]: Please use alias path like previous line

File: src/components/resources/algorithms/index.js [hunk 6]
```diff
@@ -48,111 +48,124 @@ const Description = tw.div``;
...
+  useEffect(()=>{
+    fetch("http://localhost:3001/resources/getresources",)
+      .then((res) => {
+        return (res.json());
+      })
+      .then((result)=> {
+        setPost(getArray(result,0,staticData));
+      })
+      .catch((err) => {
+        console.log(err);
```
[the remark this answers is not in the document]
Comment (ank1traj) [reply src/components/resources/algorithms/index.js:166]: Please don't use console.log

File: src/components/resources/cloud-devops/index.js [hunk 7]
```diff
@@ -48,26 +48,38 @@ const Description = tw.div``;
...
+  useEffect(()=>{
+    fetch("http://localhost:3001/resources/getresources",)
+      .then((res) => {
+        return (res.json());
+      })
+      .then((result)=> {
+        setPost(getArray(result,9,staticData));
+      })
+      .catch((err) => {
+        console.log(err);
```
Comment (ank1traj) [inline src/components/resources/cloud-devops/index.js:80]: Please don't use console.log

File: src/components/resources/getResources.js [hunk 8]
```diff
@@ -0,0 +1,58 @@
+function find(url, posts) {
+    for (let i = 0; i < posts.length; i++) {
+        if (posts[i].url == url) {
+            return i;
+        }
+    }
+    return -1;
```
[the remark this answers is not in the document]
Comment (ank1traj) [reply src/components/resources/getResources.js:7]: Why not using this suggestion?

File: src/components/resources/getResources.js [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+}
+function driveLink(posts){
+    let arr=[];
+    for (let i = 0; i < posts.length; i++) {
+            if (posts[i].url.slice(0,17) != "https://res.cloud") {
+                arr.push(i);
+        }
+    }
+    return arr;
```
[the remark this answers is not in the document]
Comment (ank1traj) [reply src/components/resources/getResources.js:16]: Same, why we're not using this suggestion? @singhcoder694

**Review by ank1traj — CHANGES_REQUESTED** [review]

**Comment by singhcoder694** [comment]
@ank1traj I will resolve all the conversation as I make the PR for #3

**Comment by ank1traj** [comment]
> @ank1traj I will resolve all the conversation as I make the PR for #3

I didn't get it.

**Comment by singhcoder694** [comment]
> > @ank1traj I will resolve all the conversation as I make the PR for #3
> 
> I didn't get it.

I am working on #3 issue and learning graphql api for GitHub so within 1 to 2 hr I will make PR for that issue. After that I will resolve these conversation and finally will make commit. I hope u understand since today is last day to make PR under CodePeak'23 after that as you will say I would make changes.

**Comment by ank1traj** [comment]
Okay, so merging this PR. Just have reference of all the things required in this PR.

**Pull request merged**